### ISSUE TITLE
Write API Docs

### ISSUE THREAD

**Issue by mntmn** [spacedeck/spacedeck-open#15]
The RESTful API should be documented, preferrably with a generator.

**Comment by huzaifahj** [comment]
I appreciate this will be in the upcoming docs but how does the API authentication work? I just need to authenticate as admin on server side, to create a new Space. Then I'll give the ID and edit hash to the correct users. I can see that I have to send a POST request to /api/spaces, but what should be in the request? Thank you

**Comment by mntmn** [comment]
This frontend JS file has all the API requests bundled:

https://github.com/spacedeck/spacedeck-open/blob/mnt/public/javascripts/backend.js

Every REST API request has to be authenticated with a token that is passed as a HTTP header:

`X-Spacedeck-Auth: api_token`

This is how you get such a token:

https://github.com/spacedeck/spacedeck-open/blob/mnt/public/javascripts/backend.js#L268

You POST a JSON to /sessions with:
```
{
  "email": ...
  "password": ...
}
```

On success, you'll get an object back that has a field `token`. This is what you pass as the `X-Spacedeck-Auth`.

**Comment by huzaifahj** [comment]
Thank you! Can I use the token forever or have to keep refreshing?

**Comment by mntmn** [comment]
Yeah AFAIK the tokens don't expire yet (this should be implemented soon, though!). You can delete a session / token manually via the API.

**Comment by huzaifahj** [comment]
[Weirdly I still get a 400 error even though I've formatted properly](https://imgur.com/a/NfrfdcJ)

**Comment by huzaifahj** [comment]
What parameters does /api/spaces need? My body is
```
{
	"name":"Example Space",
	"user": {
		"_id":"0d23f71e-77d2-4da9-94b3-d73864363ef7"
	}
}
```
The space is created and I can access it directly. However, it doesn't show up in my dashboard or at GET /api/spaces
